The default dirty ratio is chosen to be a compromise between throughput and
overall system latency. On a desktop, if an application writes to disk a lot,
that application should be the one to slow down rather than the desktop as a
whole. At higher dirty ratio settings, an application could write a lot to
disk and then happily use lots of CPU time after that while the rest of the
system is busy waiting on that naughty applications disk writes to complete
before anything else happening.

Lower ratios mean that the application that do a lot of disk writes end up
being responsible for their own actions and they're the ones that slow down
rather than the system in general.

This does decrease overall write throughput, but to the benefit of the latency
of the system as a whole with disk writes.

-ck

---
 mm/page-writeback.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-3.0.0-ck1/mm/page-writeback.c
===================================================================
--- linux-3.0.0-ck1.orig/mm/page-writeback.c	2011-08-11 12:41:05.940183573 +1000
+++ linux-3.0.0-ck1/mm/page-writeback.c	2011-08-11 12:44:12.560183568 +1000
@@ -78,7 +78,7 @@ int vm_highmem_is_dirtyable;
 /*
  * The generator of dirty data starts writeback at this percentage
  */
-int vm_dirty_ratio = 20;
+int vm_dirty_ratio = 1;
 
 /*
  * vm_dirty_bytes starts at 0 (disabled) so that it is a function of
